MovieQuiz/Services/QuestionFactory.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  QuestionFactory.swift
+//  MovieQuiz
+//
+//  Created by Андрей Парамонов on 10.01.2023.
+//
+
+import Foundation
+
+class QuestionFactory: QuestionFactoryProtocol {
+    private weak var delegate: QuestionFactoryDelegate?
+    private var current = -1
+
+    init(delegate: QuestionFactoryDelegate) {
+        self.delegate = delegate
+    }
+
+    func requestNextQuestion() {
+        current = generateNext()
+        let question = questions[safe: current]
+        delegate?.didReceiveNextQuestion(question: question)
+    }
+
+    private func generateNext() -> Int {
+        var index: Int?
+        repeat {
+            index = (0..<questions.count).randomElement()
+        } while (index == nil || index == current)
+        return index!
```

Comment:
позволю себе немного поворчать  :)
```suggestion
        var index: Int?
        repeat {
            index = (0..<questions.count).randomElement()
        } while (index == nil || index == current)
        return index ?? 0
```

Reply:
оно не будет nil из-за условия выше в цикле)
но я поправлю, чтобы не орать! на код